### PR TITLE
Consul 1.1.0 has removed the script argument for checks.

### PR DESCRIPTION
**Pull request by cablehead**
This PR also bumps the Consul test binaries to 1.1.0 and updates test assertions for agent/services calls as that now returns an addition `Meta` field.

### CODE REVIEW

File: consul/base.py [hunk 1]
```diff
@@ -49,11 +50,16 @@ class Check(object):
...
+    def script(klass, args, interval):
         """
-        Run *script* every *interval* (e.g. "10s") to peform health check
+        Run the script *args* every *interval* (e.g. "10s") to peform health
+        check
         """
-        return {'script': script, 'interval': interval}
+        if isinstance(args, six.string_types) \
+                or isinstance(args, six.binary_type):
+            log.warn("DEPRECATED: Check.script should take a list of args")
```
Comment (matusvalo) [inline consul/base.py]: Maybe here can be used proper deprecation warning from python std lib?
```python
warnings.warn("Check.script should take a list of args", DeprecationWarning)
```

**Comment by matusvalo** [comment]
One additional comment. Is it possible to somehow specify minimal version supported of Consul? Will we create separate API for it something similar to sys.version_info or we can just have a place in docs for it.

**Comment by asteven** [comment]
What I did in my cdist types (config management system) was to wrap 'script' in 'args'.
Translated to this context this could look something liket this:

```
def something(self):
    # ...
    warnings.warn("Check.script should take a list of args", DeprecationWarning)
    return {'args': ["sh", "-c", script], 'interval': interval}
```

This way you can emit the deprecation warning but you don't break the user.
Then in some future version you remove the old code.

**Comment by cablehead** [comment]
@matusvalo we've been keeping the version of python-consul correlated with the version of Consul it corresponds to. e.g. this next release will be python-consul 1.1.0. do you think that's sufficient to communicate suggested minimal version?

**Comment by cablehead** [comment]
Good call @asteven, from the Consul docs:

>  Prior to Consul 1.0, checks used a single Script field to define the command to run, and would always run in a shell. 

I'll change it to your suggestion.

**Comment by cablehead** [comment]
I got pull into a small work emergency yesterday, which has delayed this release. Still hoping to get it out this weekend. My family is vacation for the next 3 weeks from Monday so I won't be able to release anything then.

**Pull request merged**